### ISSUE TITLE
error RS2007: Analyzer release file 'AnalyzerReleases.Shipped.as a missing or invalid release header

### ISSUE THREAD

**Issue by RussKie** [dotnet/roslyn-analyzers#5481]
### Analyzer

[RS2007](https://github.com/dotnet/roslyn-analyzers/blob/main/src/Microsoft.CodeAnalysis.Analyzers/Microsoft.CodeAnalysis.Analyzers.md#rs2007-invalid-entry-in-analyzer-release-file): Invalid entry in analyzer release file

### Analyzer source

**SDK**: 6.0.100-rc.1.21416.15

### Describe the bug

Observed while publishing analyzer info in dotnet/winforms release/6.0 branch.

```md
; Shipped analyzer releases
; https://github.com/dotnet/roslyn-analyzers/blob/master/src/Microsoft.CodeAnalysis.Analyzers/ReleaseTrackingAnalyzers.Help.md

## Release .NET 6.0 RC2

### New Rules

Rule ID | Category | Severity | Notes
--------|----------|----------|--------------------
WFAC010 | ApplicationConfiguration | Warning | AppManifestAnalyzer, [Documentation](https://github.com/dotnet/winforms/blob/main/src/System.Windows.Forms.Analyzers/AppManifestAnalyzer.Help.md)

```

![image](https://user-images.githubusercontent.com/4403806/133349838-476e79dd-1eac-4ae5-999f-4517f9361635.png)

I tried removing `## Release .NET 6.0 RC2` line, then the I get an error against `### New Rules` line... Right now I'm suppressing `RS2007` but I'm puzzled by how `RS2007` is expected to work.

https://github.com/dotnet/roslyn-analyzers/issues/4484 appears to discuss a similar issue but I didn't see anything related to this scenario.

**Comment by mavasani** [comment]
Analyzer currently expects the release header to be in a format `## Release <%Version%>`: https://github.com/dotnet/roslyn-analyzers/blob/32658a620ba89e963ea2bcc39554bec06b1d654b/src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/ReleaseTrackingHelper.cs#L58-L73

It seems possible to enhance the analyzer to expect other formats, that would be a new feature request.

**Comment by glen-nicol** [comment]
I'm getting this same error for the following file:

```markdown
; Shipped analyzer releases
; https://github.com/dotnet/roslyn-analyzers/blob/main/src/Microsoft.CodeAnalysis.Analyzers/ReleaseTrackingAnalyzers.Help.md

## Release 0.1.0

### New Rules

Rule ID | Category        | Severity | Notes
--------|-----------------|----------|--------------------
RUI001  | Design          |  Error   |
RUI002  | Design          |  Error   |
```


And the project has ```<Version>0.1.0</Version>```

Why does the error not mention that header format and instead quotes the table?

**Comment by AArnott** [comment]
I'm getting this warning too:

> warning RS2007: Analyzer release file 'AnalyzerReleases.Shipped.md' has a missing or invalid release header '## Release 0.1.705-alpha' (https://github.com/dotnet/roslyn-analyzers/blob/main/src/Microsoft.CodeAnalysis.Analyzers/ReleaseTrackingAnalyzers.Help.md)

Apparently it doesn't like prerelease syntax.

Workaround:

```diff
-## Release 0.1.705-alpha
+## Release 0.1.705
```

**Comment by wachidsusilo** [comment]
It demands the string to be exactly start with these:
https://github.com/dotnet/roslyn-analyzers/blob/32658a620ba89e963ea2bcc39554bec06b1d654b/src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/ReleaseTrackingHelper.cs#L29C1-L30C109

while this format is should be fine:

![Image](https://github.com/user-attachments/assets/22c9f990-528d-4a69-9b4f-5949132ef2e6)

**Comment by AVee** [comment]
And to make matters worse, the [linked file](https://github.com/dotnet/roslyn-analyzers/blob/main/src/Microsoft.CodeAnalysis.Analyzers/ReleaseTrackingAnalyzers.Help.md)  in both the template and the resulting warning doesn't even exitst anymore.

**Comment by ViRuSTriNiTy** [comment]
Regarding the broken "linked file" link posted by @AVee:

The current documentation can be found here: 

https://github.com/dotnet/roslyn/blob/469f6d9cf08b3209f11459119a9fd3afafc65494/src/RoslynAnalyzers/Microsoft.CodeAnalysis.Analyzers/ReleaseTrackingAnalyzers.Help.md

**Comment by rcocks-hl** [comment]
> It demands the string to be exactly start with these: https://github.com/dotnet/roslyn-analyzers/blob/32658a620ba89e963ea2bcc39554bec06b1d654b/src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/ReleaseTrackingHelper.cs#L29C1-L30C109
> 
> while this format should be fine:
> 
> ![Image](https://github.com/user-attachments/assets/22c9f990-528d-4a69-9b4f-5949132ef2e6)

Thank you, although now I have to suppress a markdown linter warning that wants to put the pipes at the start and end, at least that one isn't blocking the build!
